basic_bootloader/src/bootloader/gas_helpers.rs

```diff
@@ -7,10 +7,17 @@ use zk_ee::system::{Computational, Resources};
 
 use super::*;
 
-/// Returns the resources for the transaction and the withheld resources.
-/// The withheld resources are the resources that are withheld from the transaction's
-/// execution to ensure that it does not use too many native computational resources.
-/// They are reclaimed at the end of the transaction and used to charge the pubdata.
+pub struct ResourcesForTx<S: EthereumLikeTypes> {
+    // Resources to run the transaction
+    pub main_resources: S::Resources,
+    /// Resources that are withheld from the transaction's
+    /// execution to ensure that it does not use too many native computational resources.
```

Comment:
Maybe smth like "can't spend more then MAX_NATIVE_COMPUTATIONAL computational resources. These excessive resources still can be used to pay for pubdata"?

Reply:
Reworded it